### PR TITLE
sia-demo

### PR DESCRIPTION
**Pull request by jabadge**
I updated the penalty term of the shallow_ice model. I also added a new demo using it to model a synthetic ice sheet. Because this demo is simpler than all but the 00-meshes-functions demo, I rearranged the order of the demos and adjusted the wording of the synthetic-ice-shelf demo to reflect this new ordering.

### CODE REVIEW

**Comment by danshapero** [comment]
So it looks like you're getting a test failure because now you need to pass the thickness when calculating the scale functional as well as the velocity. The line that's causing the test failure is [here](https://github.com/icepack/icepack/blob/b1552d1e04469f8545845b88cf2ae91735f3914f/test/shallow_ice_test.py#L111). You could just add `thickness=h` as another keyword argument to that function call to fix it.

After that's fixed, we'll get another failure because the test that you wrote before assumed very minimal smoothing and thus expected an asymptotic convergence rate close to 1. Now as we've both learned from you writing this demo, using that little smoothing is impractical once you actually try to do anything interesting. But it also means that we're no longer solving the equations exactly and so we shouldn't expect the same convergence rate. You can change [this line](https://github.com/icepack/icepack/blob/b1552d1e04469f8545845b88cf2ae91735f3914f/test/shallow_ice_test.py#L118) to assert that the slope is greater than 0 instead of 0.9. We can think of a better testing strategy later but this is fine for now.

I've made some more specific comments as reviews, but those are minor and all together this looks great!

File: demo/01-synthetic-ice-sheet.ipynb [hunk 1]
```diff
@@ -0,0 +1,631 @@
...
+   "source": [
+    "The main loop to run the simulation consists of alternating calls to the prognostic and diagnostic solve methods of the model object.\n",
+    "We'll see this pattern repeated in all of the demos that follow.\n",
+    "In some cases we'll add in extra physics, like updating the temperature or damage field, but the core idea is the same.\n",
+    "\n",
+    "**All of the fields involved in this simulation -- the ice thickness, velocity, and so forth -- were passed by keyword.**\n",
+    "Only the timestep `dt` was passed by position.\n",
+    "This is true more broadly of every solving routine in icepack.\n",
+    "Passing arguments by keyword is admittedly more verbose than passing them by position, but we've made this choice for an important reason.\n",
+    "As we'll show later on in this demo, passing arguments by keyword makes it possible to swap out components of the model physics for your own customized versions.\n",
```
Comment (danshapero) [inline demo/01-synthetic-ice-sheet.ipynb:480]: This should be changed from "later on in this demo" to "in the next demo".

File: demo/01-synthetic-ice-sheet.ipynb [hunk 2]
```diff
@@ -0,0 +1,631 @@
...
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "fig, axes = icepack.plot.subplots()\n",
+    "contours = icepack.plot.tricontourf(firedrake.interpolate(s,Q), 40, axes=axes, cmap='magma')\n",
```
Comment (danshapero) [inline demo/01-synthetic-ice-sheet.ipynb:519]: This interpolate call should be unnecessary too.

**Review by danshapero — CHANGES_REQUESTED** [review]
Mostly minor things, there's some wordage left over from the ice shelf demo that should say "ice sheet" instead, and a handful of calls to `interpolate` are redundant.

**Comment by jabadge** [comment]
@danshapero Thanks for these edits! I made your recommended changes to the ice sheet demo and shallow ice test.

**Review by danshapero — APPROVED** [review]

**Pull request merged**